src/Pact/Runtime/Capabilities.hs

```diff
@@ -146,19 +167,21 @@ evalUserCapability i af scope cap cdef test = go scope
     -- Callstack: check if managed, in which case push/emit,
     -- otherwise push and test.
     evalStack = checkManaged i af cap cdef >>= \r -> case r of
-      Nothing -> push >> test >> emitMaybe
-      Just composed -> emitCap >> pushSlot (CapSlot scope cap composed)
+      Nothing -> push >> test >> emitMaybe >> popCapStack installFinished
+      Just composed -> emitCap >> pushSlot (CapSlot scope False cap composed)
+
+    installFinished c = evalCapabilities . capStack %= (c { _csEvaluating = False} :)
 
     -- Composed: check if managed, in which case install onto head/emit,
     -- otherwise push, test, pop and install onto head
     evalComposed = checkManaged i af cap cdef >>= \r -> case r of
       Nothing -> push >> test >> emitMaybe >> popCapStack installComposed
-      Just composed -> emitCap >> installComposed (CapSlot scope cap composed)
+      Just composed -> emitCap >> installComposed (CapSlot scope False cap composed)
```

Comment:
Why are `composed` not considered evaluating??

Reply:
Reverted